### ISSUE TITLE
Getting Error While running COAWST (ROMS+SWAN+ST) 

### ISSUE THREAD

**Issue by krprakash2** [DOI-USGS/COAWST#46]
Hi, 

While running COAWST (ROMS+SWAN+ST)  I am getting some wrong value (temp, salt, u & v) in the output at particular area in 1st time step of writing output. The model is running up to 18 days after that it's  blown up. I tried my ways e.g. by  reducing time step, by changing model physics and by suing ROMS alone configuration in all the case error still remain same. Also, I checked all the inputs all seems fine. 

Here I am attaching the plot of temperature, u, v and cppdef file.

![temp](https://user-images.githubusercontent.com/85049236/120093099-8abcf980-c135-11eb-8eea-fe18714a9262.gif)
![u](https://user-images.githubusercontent.com/85049236/120093102-901a4400-c135-11eb-93da-2a481684cfa7.gif)
![V](https://user-images.githubusercontent.com/85049236/120093105-95778e80-c135-11eb-8053-fe9c3d9d52b2.gif)

[sedmclim.txt](https://github.com/jcwarner-usgs/COAWST/files/6565755/sedmclim.txt)

In the log file Metrics information for Grid 01:
 ===============================

 Minimum X-grid spacing, DXmin =  1.05693151E+00 km
 Maximum X-grid spacing, DXmax =  1.07346081E+00 km
 Minimum Y-grid spacing, DYmin =  1.05701699E+00 km
 Maximum Y-grid spacing, DYmax =  1.07343656E+00 km
 Minimum Z-grid spacing, DZmin =  1.25324381E-02 m
 Maximum Z-grid spacing, DZmax =  2.50833018E+02 m

 Minimum barotropic Courant Number =  5.52708204E-04
 Maximum barotropic Courant Number =  3.20683344E-02
 Maximum Coriolis   Courant Number =  1.79786245E-04

 Minimum horizontal diffusion coefficient =  8.80811871E+00 m2/s
 Maximum horizontal diffusion coefficient =  8.94540572E+00 m2/s

 Minimum horizontal viscosity coefficient =  8.80811871E+00 m2/s
 Maximum horizontal viscosity coefficient =  8.94540572E+00 m2/s

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Basin information for Grid 01:

 Maximum grid stiffness ratios:  rx0 =   2.258206E-01 (Beckmann and Haidvogel)
                                 rx1 =   7.696672E+00 (Haney)

So could you please suggest regarding the issue.

With Regards,
Ravi Prakash

**Comment by jcwarner-usgs** [comment]
what is happening along the southern boundary?  what does the bath look like? how are you forcing temp and salt along south boundary?

**Comment by krprakash2** [comment]
The bathymetry looks like as:
![image](https://user-images.githubusercontent.com/85049236/120341896-ebeef380-c314-11eb-8b41-0c09de63f79e.png)

Temp and salt along the south boundary at surface looks like:

![temp](https://user-images.githubusercontent.com/85049236/120343731-9582b480-c316-11eb-957a-7bbcbac619e7.gif)
![salt](https://user-images.githubusercontent.com/85049236/120343761-9b789580-c316-11eb-8ae0-9b54a527f88b.gif)

the boundary condition which I used:
LBC(isFsur) ==   Clo      Che     Che     Clo 
                    Clo      Che     Che     Clo        ! free-surface
   LBC(isUbar) ==   Clo      Shc     Shc     Clo 
                    Clo      Shc     Shc     Clo        ! 2D U-momentum
   LBC(isVbar) ==   Clo      Shc     Shc     Clo 
                    Clo      Shc     Shc     Clo        ! 2D V-momentum
   LBC(isUvel) ==   Clo      Cla     Cla     Clo 
                    Clo      Cla     Cla     Clo        ! 3D U-momentum
   LBC(isVvel) ==   Clo      Cla     Cla     Clo 
                    Clo      Cla     Cla     Clo        ! 3D V-momentum
   LBC(isMtke) ==   Clo      Red     Red     Clo 
                    Clo      Red     Red     Clo        ! mixing TKE

   LBC(isTvar) ==   Clo      Cla     Cla     Clo        ! temperature
                    Clo      Cla     Cla     Clo  
                    Clo      Cla     Cla     Clo        ! temperature
                    Clo      Cla     Cla     Clo       ! salinity

**Comment by jcwarner-usgs** [comment]
LBC(isFsur) == Clo Che Che Clo
Clo Che Che Clo ! free-surface
is this a nested application?
if so it needs to be

LBC(isFsur) == Clo Che Che Clo
Nes  Nes Nes Nes  ! free-surface

for all the ubar, vbar, u, v, temp, etc.

**Comment by krprakash2** [comment]
No it's not nested application. But is that make problem?

**Comment by krprakash2** [comment]
I commented the 2nd line still getting same problem

**Comment by jcwarner-usgs** [comment]
the bc's you are using:

LBC(isFsur) == Clo Che Che Clo
LBC(isUbar) == Clo Shc Shc Clo
LBC(isVbar) == Clo Shc Shc Clo
LBC(isUvel) == Clo Cla Cla Clo
LBC(isVvel) == Clo Cla Cla Clo
LBC(isTvar) == Clo Cla Cla Clo 
Clo Cla Cla Clo

suggest you try some other options for the south. perhaps the clamped u, v, and temp are giving an issue.  maybe try radiation or other, and see if that makes a difference.  you will need to try a few things here.

**Comment by krprakash2** [comment]
I have tried with multiple combination but each time model is blown up after the same time step.

**Comment by jcwarner-usgs** [comment]
how about you just try ROMS by itself.
if it blows up after 18 days, then you need to look at where the instability is coming from.  is it that spot on the southern boundary? then something is causing that.  you need to figure this out, it is difficult for me to do that.
does the temp at that location slowly get to that strange value?  the temp and salt should be forcing over the water depth. look at the complete data.  you are clamping u and v, what do they look like?

**Comment by krprakash2** [comment]
yes I had tried ROMS alone. And when I was running with GLS mixing it ran for 18 day with the spot coming from 1st day (even at stating time step) But when I am runing with LMD it falis to run after few time step (< 10 min).
Now I am playing with THETA_B, THETA_S, Vstretching and Vtransform. It is significant now the spot at southern boundary is not coming but now some higher value of U & V coming along north side of coast (1.5 m/s) after 10 min of run,  model is still running.

The U current looks like after few time step

<img width="809" alt="Screenshot 2021-06-02 at 7 46 21 PM" src="https://user-images.githubusercontent.com/85049236/120496536-4488c400-c3db-11eb-83d6-4926a2cb02ad.png">

**Comment by jcwarner-usgs** [comment]
has this issue been resolved?

**Comment by kravi1220** [comment]
No it has not been resolved yet. Actually I have not tried after that.
